analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/HashesShouldHaveUnpredictableSalt.cs

```diff
@@ -139,46 +139,33 @@ public void Conditional(int arg, string password)
             rng.GetBytes(salt);
             new PasswordDeriveBytes(passwordBytes, salt);           // Compliant
         }
-        new PasswordDeriveBytes(passwordBytes, salt);               // FIXME Non-compliant {{Make this salt unpredictable.}}
+        new PasswordDeriveBytes(passwordBytes, salt);               // Noncompliant {{Make this salt unpredictable.}}
 
         var noncompliantSalt = new byte[16];
         var compliantSalt = new byte[16];
         var salt3 = arg == 2 ? compliantSalt : noncompliantSalt;
```

Comment:
aren't both noncompliant here? I don't see the difference.

Reply:
If `arg == 1` then the `RandomNumberGenerator` will fill up the salt with cryptographically strong random numbers, therefore it's compliant.
Otherwise, it will remain non-random, hence Noncompliant.

Reply:
Maybe this example should turn to 
```csharp
var noncompliantSalt = new byte[5];
var compliantSalt = new byte[16];
rng.GetBytes(compliantSalt );
var salt3 = arg == 2 ? compliantSalt : noncompliantSalt;
```
also please test the opposite where the compliant is passed.